test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_full_dispatch.cpp

```diff
@@ -0,0 +1,367 @@
+        work_offset = 42;
+        num_elements /= 2;
+        group_size /= 2;
+
+        cl_mutable_dispatch_arg_khr arg0{ 0, sizeof(cl_mem), &in_buf_update };
+        cl_mutable_dispatch_arg_khr arg1{ 1, sizeof(svm_buffers.newWrapper),
+                                          svm_buffers.newWrapper };
+
+        cl_mutable_dispatch_exec_info_khr exec_info{};
+        exec_info.param_name = CL_KERNEL_EXEC_INFO_SVM_PTRS;
+        exec_info.param_value_size = sizeof(svm_buffers.newBuffer);
+        exec_info.param_value = &svm_buffers.newBuffer;
+
+        cl_mutable_dispatch_config_khr dispatch_config{};
+        dispatch_config.type = CL_STRUCTURE_TYPE_MUTABLE_DISPATCH_CONFIG_KHR;
+        dispatch_config.command = command;
+        dispatch_config.num_args = 1;
+        dispatch_config.arg_list = &arg0;
+        dispatch_config.num_svm_args = 1;
+        dispatch_config.arg_svm_list = &arg1;
+        dispatch_config.num_exec_infos = 1;
+        dispatch_config.exec_info_list = &exec_info;
+        dispatch_config.global_work_offset = &work_offset;
+        dispatch_config.global_work_size = &num_elements;
+        dispatch_config.local_work_size = &group_size;
+
+        cl_mutable_base_config_khr mutable_config{
+            CL_STRUCTURE_TYPE_MUTABLE_BASE_CONFIG_KHR, nullptr, 1,
+            &dispatch_config
+        };
+
+        error = clUpdateMutableCommandsKHR(command_buffer, &mutable_config);
+        test_error(error, "clUpdateMutableCommandsKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        // Check the results of the modified execution
+        auto check_info_result = [&](const cl_uint param, const size_t test) {
+            size_t info_res = 0;
+            error = clGetMutableCommandInfoKHR(command, param, sizeof(info_res),
+                                               &info_res, nullptr);
+            test_error_ret(error, "clGetMutableCommandInfoKHR failed", false);
+
+            if (info_res != test)
+            {
+                log_error("ERROR: Wrong value returned from "
+                          "clGetMutableCommandInfoKHR.");
+                return false;
+            }
+            return true;
+        };
+
+        if (!check_info_result(CL_MUTABLE_DISPATCH_GLOBAL_WORK_SIZE_KHR,
+                               num_elements))
+            return TEST_FAIL;
+        if (!check_info_result(CL_MUTABLE_DISPATCH_GLOBAL_WORK_OFFSET_KHR,
+                               work_offset))
+            return TEST_FAIL;
+        if (!check_info_result(CL_MUTABLE_DISPATCH_LOCAL_WORK_SIZE_KHR,
+                               group_size))
+            return TEST_FAIL;
+
+        if (!verify_result(svm_buffers.newBuffer, group_size, work_offset))
+            return TEST_FAIL;
+
+        error = clFinish(queue);
```

Comment:
suggest moving this to after `clEnqueueCommandBufferKHR`, I'd expect us to wait on the device work to complete and then start doing the verification steps.

Reply:
Done